src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4787,8 +4766,8 @@ private PostSkipAction SkipBadVariableListTokens(SeparatedSyntaxListBuilder<Vari
             CSharpSyntaxNode tmp = null;
             Debug.Assert(list.Count > 0);
             return this.SkipBadSeparatedListTokensWithExpectedKind(ref tmp, list,
-                p => p.CurrentToken.Kind != SyntaxKind.CommaToken,
-                p => p.CurrentToken.Kind == SyntaxKind.SemicolonToken || p.IsTerminator(),
+                static p => p.CurrentToken.Kind != SyntaxKind.CommaToken,
+                static (p, _) => p.CurrentToken.Kind == SyntaxKind.SemicolonToken || p.IsTerminator(),
```

Comment:
I didn't understand why we don't need to pass a `closeKind` here, but we do for parameter lists, for example.

Reply:
So, variable lists only stop once you hit a `;`.  In this case, because this code knows that, it just inlines teh close token it expects right there.  It would be just as correct to pass in SyntaxKind.SemicolonToken in as the closeTokenKind, and then check it in the lambda.  I've made that more consistent for all the other comma-separated cases i fixed up.